modules/core/src/shadertools/src/modules/picking/picking.js

```diff
@@ -77,7 +77,18 @@ const float COLOR_SCALE = 1. / 255.;
  */
 vec4 picking_filterHighlightColor(vec4 color) {
   bool selected = bool(picking_vRGBcolor_Aselected.a);
-  return selected ? (picking_uHighlightColor * COLOR_SCALE) : color;
+
+  if (selected) {
+    vec4 highLightColor = picking_uHighlightColor * COLOR_SCALE;
+
+    float resultAlpha = highLightColor.a + color.a * (1.0 - highLightColor.a);
+
+    vec3 resultRGB = (highLightColor.rgb * highLightColor.a
+     + color.rgb * color.a * (1.0 - highLightColor.a)) / resultAlpha;
+    return vec4(resultRGB, color.a);
```

Comment:
Should we use `highlightColor.a` as final alpha instead of `color.a`. Reason being, this only affects the object that is currently picked, so we let highlightColor define what its alpha should be. For instance, if user sets highlightColor.a = 1.0, we make highlighted object opaque ?   Other than this concern, all look good to me.

Reply:
> Should we use highlightColor.a as final alpha instead of color.a?

Makes sense to me. Either way I think we should document the chosen behavior, in the `picking` shader module docs and probably also in the deck.gl docs.

A test case would also be really cool, not sure if that is feasible with new the shader test setups, need to take a closer look at that...

Reply:
I don't think `highlightColor.a` should be used. Say I only want to "slightly tint" the object when it's highlighted, `highlightColor.a` is the only way I can control that, but I don't want it to become translucent.

This is how kepler is implementing its own picking module:
https://github.com/uber/kepler.gl/blob/master/src/shaderlib/picking-module.js#L41

Reply:
In the following two images, I set the highlight color to [0, 255, 255, 50]. The first image uses the alpha value of the original object. When the planet is picked, the blending result can show the original object with slight highlight color.  The second image uses the alpha value of the highlight color. After blending, the planet becomes almost transparent because its alpha value is 50 now.

![object_samll_alpha](https://user-images.githubusercontent.com/42359372/49537999-42336680-f87f-11e8-94cd-304cecdbb5a2.png)

![highlight_samll_alpha](https://user-images.githubusercontent.com/42359372/49538044-5bd4ae00-f87f-11e8-9039-43fb79341cce.png)

Reply:
![highlight_samll_alpha](https://user-images.githubusercontent.com/42359372/49538368-182e7400-f880-11e8-80bf-f34693f8143b.png)